plugins/woocommerce/includes/wc-template-functions.php

```diff
@@ -3214,7 +3219,11 @@ function woocommerce_form_field( $key, $args, $value = null ) {
 			$field_html = '';
 
 			if ( $args['label'] && 'checkbox' !== $args['type'] ) {
-				$field_html .= '<label for="' . esc_attr( $label_id ) . '" class="' . esc_attr( implode( ' ', $args['label_class'] ) ) . '">' . wp_kses_post( $args['label'] ) . $required_indicator . '</label>';
+				if ( $is_hidden_field ) {
+					$field_html .= '<label class="' . esc_attr( implode( ' ', $args['label_class'] ) ) . '">' . wp_kses_post( $args['label'] ) . $required_indicator . '</label>';
+				} else {
+					$field_html .= '<label for="' . esc_attr( $label_id ) . '" class="' . esc_attr( implode( ' ', $args['label_class'] ) ) . '">' . wp_kses_post( $args['label'] ) . $required_indicator . '</label>';
+				}
```

Comment:
instead of if/else why don't we construct a $maybe_for_attr and either make it an empty string, or the for attribute depending on whether the field is hidden?